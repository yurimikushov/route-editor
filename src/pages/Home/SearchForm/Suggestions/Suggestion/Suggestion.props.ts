import { Suggestion } from 'services/suggestions'

type SuggestionProps = {
  className?: string
  isHighlighted: boolean
  suggestion: Suggestion
  testId?: string
  onMouseDown: () => void
}

export default SuggestionProps
