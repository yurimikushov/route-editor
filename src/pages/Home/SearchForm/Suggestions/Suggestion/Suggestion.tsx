import { FC } from 'react'
import cn from 'classnames'
import SuggestionProps from './Suggestion.props'

const Suggestion: FC<SuggestionProps> = ({
  className,
  isHighlighted,
  suggestion,
  testId,
  onMouseDown,
}) => {
  const { name, description } = suggestion

  return (
    <li
      className={cn(
        className,
        'px-2 py-1',
        'flex flex-col',
        !isHighlighted ? 'bg-white' : 'bg-gray-50',
        'cursor-pointer',
        'hover:bg-gray-100'
      )}
      data-testid={testId}
      onMouseDown={onMouseDown}
    >
      <span className='text-sm text-black'>{name}</span>
      <span className='text-xs text-gray-400'>{description}</span>
    </li>
  )
}

export default Suggestion
