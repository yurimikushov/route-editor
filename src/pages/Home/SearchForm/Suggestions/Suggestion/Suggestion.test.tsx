import { fireEvent, render, screen } from '@testing-library/react'
import renderer from 'react-test-renderer'
import noop from 'lodash/noop'
import Suggestion from './Suggestion'

it('should render highlighted suggestion', () => {
  const tree = renderer
    .create(
      <Suggestion
        className='some-class-name'
        isHighlighted
        suggestion={{
          name: 'Balalyaka',
          description: 'Some description',
          point: {
            lat: 0,
            lon: 0,
          },
        }}
        onMouseDown={noop}
      />
    )
    .toJSON()
  expect(tree).toMatchSnapshot()
})

it('should render non-highlighted suggestion', () => {
  const tree = renderer
    .create(
      <Suggestion
        className='some-class-name'
        isHighlighted={false}
        suggestion={{
          name: 'Balalyaka',
          description: 'Some description',
          point: {
            lat: 0,
            lon: 0,
          },
        }}
        onMouseDown={noop}
      />
    )
    .toJSON()
  expect(tree).toMatchSnapshot()
})

it('should call onMouseDown when is mouse down', () => {
  let isClicked = false
  const handleMouseDown = jest.fn(() => {
    isClicked = true
  })

  render(
    <Suggestion
      className='some-class-name'
      isHighlighted
      suggestion={{
        name: 'Balalyaka',
        description: 'Some description',
        point: {
          lat: 0,
          lon: 0,
        },
      }}
      testId='suggestion'
      onMouseDown={handleMouseDown}
    />
  )

  fireEvent.mouseDown(screen.getByTestId('suggestion'))

  expect(isClicked).toBeTruthy()
  expect(handleMouseDown).toHaveBeenCalledTimes(1)
})
