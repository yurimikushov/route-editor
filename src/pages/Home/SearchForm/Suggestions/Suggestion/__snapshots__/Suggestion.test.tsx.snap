// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render highlighted suggestion 1`] = `
<li
  className="some-class-name px-2 py-1 flex flex-col bg-gray-50 cursor-pointer hover:bg-gray-100"
  onMouseDown={[Function]}
>
  <span
    className="text-sm text-black"
  >
    Balalyaka
  </span>
  <span
    className="text-xs text-gray-400"
  >
    Some description
  </span>
</li>
`;

exports[`should render non-highlighted suggestion 1`] = `
<li
  className="some-class-name px-2 py-1 flex flex-col bg-white cursor-pointer hover:bg-gray-100"
  onMouseDown={[Function]}
>
  <span
    className="text-sm text-black"
  >
    Balalyaka
  </span>
  <span
    className="text-xs text-gray-400"
  >
    Some description
  </span>
</li>
`;
